### ISSUE TITLE
[GENERAL] WRONG-INCONSISTENT GRAPHICS

### ISSUE THREAD

**Issue by panagiotissp** [labooltd/GameAndroid#18]
This is going to be a big list of  wrong graphics, or inconsistent graphics that may finally affect UX,
either directly - e.g. elements not aligned correctly as it may happen currently in the Categories page - 
or indirectly - e.g. be responsible for small movements of elements when an event is triggered.

I'll put below the list with the wrong graphics and possible inconsistencies I have found, but this mostly for customer to review and decide if it is sth that is actually needed to be changed.

**Comment by panagiotissp** [comment]
WRONG GRAPHICS BELOW - ie images that I understood that are problematic by comparison with another image:
(All dimensions below are in pixels)

1. Main menu:
- btn_fb: 161x160 Should be 160x160 (as the rest of small bottom icons)
- logo: 600x827px Too much empty space on the top. I have a custom fixed version of 600x744 (this is used in the app for demo purposes) but they should provide theirs.
- popup images are missing (triangle image, any internal images etc)

2. GameSettingsP1:
- All selected circular images are 104x103 (just mentioned since common sense tells that they should be 104x104 or 103x103)
- All unselected circular images are 97x96 except the blue one which is 97x97

- top buttons, grouped by different states:
Button '2' unselected: 188x185
Button '3' unselected: 189x185
Button '4' unselected: 188x185

Button '2' selected: 197x245
Button '3' selected: 198x245
Button '4' selected: 197x245

- Also, top buttons have inconsistency among them.
Width dimension should be the same to all buttons, either selected or unselected

3. GameSettingsP2:
- toggle_off_img: 169x99
toggle_on_img: NOT PROVIDED (temporarily using toggle off image created by me for demo purposes, using the 169x99 dimensions of toggle_off img)


- top buttons grouped by state:
Button '  5' unselected: 205x203
Button '10' unselected: 206x203
Button '15' unselected: 206x203
Button '20' unselected: 205x203

Button '  5' selected: 216x271
Button '10' selected: 488x279
Button '15' selected: 216x278
Button '20' selected: 216x278

- top buttons have inconsistency among them. 
Width dimension should be the same to all buttons, either selected or unselected

4. Categories Page:
- If it is wanted the category items to take all the same vertical space on the screen then this probably means that the icons provided for each category item should be of the same height (in order for them not to create different heights per category) which is not the case at all.
- If it is wanted the category items to take all the same horizontal space on the screen then this probably means that the icons provided for each category item should be of the same width (in order for them not to create different widths per category or for not just 1 icon with big width to push the texts of all categories far away on the right) which is not the case at all.

_So, if they have any objection with how the ui looks for that page then it would probably be best to check these icons' dimensions in comparison._

5. Go Page:
- In Regular  mode, go_img is 773x772
In Football mode, go_img is 813x812

6. Questions Page:
- Header backgroun images (all 4 of them) have a white space at the top right corner
- Clock img: NOT PROVIDED
- In order to have the progress bar effect, an image is wanted that displays just the clock and the inside being completely transparent except from the small lines (these that are separating the progress bar per 7.5 seconds) that should exist. (currently an image created by me is used for the demo purposes)

7. Attack Bonus Page:
- joker_give_question_blue: 215x218
joker_give_question_cyan: 216x218
joker_give_question_magenta: 215x218
joker_give_question_orange: 215x219

- joker_multiplier_blue: 215x218
joker_multiplier_cyan: 216x219
joker_multiplier_magenta: 215x219
joker_multiplier_orange: 215x219

- joker_switch_question_blue: 215x218
joker_switch_question_cyan: 216x218
joker_switch_question_magenta: 215x218
joker_switch_question_orange: NOT PROVIDED

- joker_add_two_blue: 215x218
joker_add_two_cyan: NOT PROVIDED
joker_add_two_magenta: NOT PROVIDED
joker_add_two_orange: NOT PROVIDED

- joker_locked_img: 239x275 (generally it should be same dimensions as unlocked images. May create inconsistencies)

8. Defence-Evil Bonus Page:
- joker_all_one_blue: 217x220
joker_ all_one _cyan: 216x219
joker_ all_one _magenta: 217x221
joker_ all_one _orange: 217x220

- joker_decrease_time_blue: 217x219
joker_ decrease_time _cyan: 216x219
joker_ decrease_time _magenta: 217x220
joker_ decrease_time _orange: 217x219

- joker_divider_blue: 217x219
joker_divider_question_cyan: 216x219
joker_divider_question_magenta: 217x220
joker_divider_question_orange: 217x219

- joker_locked_img: 239x275 (generally it should be same dimensions as unlocked images. May create inconsistencies)

9. Scores Page:
- Title img: 226x226. Sudden death corresp img though: 259x258

10. Sudden Death Page:
- Title img: 259x258 . Scores page corresp img though:  226x226 as said above
- Img displayed instead of 'Round' text: NOT PROVIDED

11. Winner Page:
- Confetti img in normal mode: 1863x1577
Confetti img in football mode: 1170x1634

- Medal img (normal mode): 223x309
Ball img (football mode): 239x235

- Champagne img (normal mode): 499x629
Cup img (football mode): 593x617

12. Options Page:
- ads_off img: 109x156
ads_on img: 108x156

- sounds_off img: 111x85
sounds_on img: 106x97

- stats_off img: 112x98
stats_on img: 103x98

- toggle_off img: 258x143
toggle_on img: 249x138

- help popup triangle img: NOT PROVIDED

13. Store Page:
- Football icon: 134x134
jokers icon (yellow): 134x135
jokers icon (grey): 134x135
no_ads icon: 134x135
premium_icon: 132x131

**Comment by ignacioch** [comment]
Thanks for the list, will pass it to them today and update the issue with the new graphics once I have them.

**Comment by panagiotissp** [comment]
If it will take too long for the whole list, then please send mini updates per page, that's why I put them in a numbered list.

**Comment by ignacioch** [comment]
Images send to client. Waiting for them to send them back. Will update with an ETA.

**Comment by ignacioch** [comment]
One question I got : 

"when you mention 
Store Page:
Football icon: 134x134
jokers icon (yellow): 134x135
jokers icon (grey): 134x135
no_ads icon: 134x135
premium_icon: 132x131"

is this the one you want or the one it is and you want them to be exact squares?

**Comment by panagiotissp** [comment]
I have no jurisdiction to "want specific graphics' dimensions".
 
All the above are the dimensions of images I currently have, and which may create inconsistencies from my understanding (but maybe not, of course).
E.g. I consider to be an inconsistency that round buttons '5', '10' are not of exactly the same dimensions.
This may - or may not - create an inconsistency of e.g. part of the layout to move up and down when selecting one of the buttons.

If customer considers there is no issue at all, then we can just close this github issue and I'll proceed with using the current graphics.

**Comment by ignacioch** [comment]
Got the graphics. https://www.dropbox.com/sh/3f3mjcuwijxl7bp/AACVXazXGXO2IZYmJ34ZFIk1a?dl=0

Let me know if they are not ok.

**Comment by ignacioch** [comment]
One note : for the toggles can we use the Android system ones and make their colour cyan when toggled? I can give you the cyan code if needed.

**Comment by panagiotissp** [comment]
I still have not checked your question in android.
However, I had a look at the graphics sent and I do not see there any of the images that I have mentioned above as 'not provided' (this is just an example of a quick check that I did, if I got graphics that currently do not exist in the app). 
Am I missing something?

**Comment by ignacioch** [comment]
- btn_fb: Provided on the link above 
- logo : link above
- popup images are missing : use android default popup.

**Game Settings P1 :**
- All selected circular images are 104x103 : link above
- Unselected : 
_Client => Leave them as it is :_
Button '2' unselected: 188x185
Button '3' unselected: 189x185
Button '4' unselected: 188x185
Button '2' selected: 197x245
Button '3' selected: 198x245
Button '4' selected: 197x245

**Game Settings P2 :**

-- Link above --
Button '2' unselected: 188x185
Button '3' unselected: 189x185
Button '4' unselected: 188x185
Button '2' selected: 197x245
Button '3' selected: 198x245
Button '4' selected: 197x245

toggle on/off : use android default - just change the background colour to be the cyan one.

**Categories**

_Leave it as it - client has no objection so far._

**Go Page**
_Leave it as it_ 

_New link_ when mentioned below :


**Questions**
- Header backgroun images : provided on new link
- Clock image : Link above. Dont worry about the separating lines. On IOS I am using a code found online with bezzier paths and splits the circle into 60 equal parts and every time it tries to fill in 1/60th of it. There should be something similar on android. If not I can send you the ios code to take a look.

** Jokers attack ** **this is a phase 2 deliverable**

_This joker has been deleted - leave it always locked for now_
joker_give_question_blue: 215x218
joker_give_question_cyan: 216x218
joker_give_question_magenta: 215x218
joker_give_question_orange: 215x219

_On the new link as doublePointsCyan etc_
joker_multiplier_blue: 215x218
joker_multiplier_cyan: 216x219
joker_multiplier_magenta: 215x219
joker_multiplier_orange: 215x219

_On the new link as switchBlue etc_
joker_switch_question_blue: 215x218
joker_switch_question_cyan: 216x218
joker_switch_question_magenta: 215x218
joker_switch_question_orange: NOT PROVIDED

_On the new link Plus2Cyan etc_
joker_add_two_blue: 215x218
joker_add_two_cyan: NOT PROVIDED
joker_add_two_magenta: NOT PROVIDED
joker_add_two_orange: NOT PROVIDED

_New link as LockedButton _
joker_locked_img

** Defence-Evil Bonus Page **

_On the new link as 1PtsBlue etc_
joker_all_one_blue: 217x220
joker_ all_one cyan: 216x219
joker all_one magenta: 217x221
joker all_one _orange: 217x220

_On the new link as Minus15SecBlue etc_
joker_decrease_time_blue: 217x219
joker_ decrease_time cyan: 216x219
joker decrease_time magenta: 217x220
joker decrease_time _orange: 217x219

_On the new link as Divided etc_
joker_divider_blue: 217x219
joker_divider_question_cyan: 216x219
joker_divider_question_magenta: 217x220
joker_divider_question_orange: 217x219

_sent above_
joker_locked_img: 239x275 (generally it should be same dimensions as unlocked images. May create inconsistencies)

** Sudden death **

They haven't send me this - will ask again.
SuddendeathRounds is provided on the new link.

** Winner Page**

_Client => Leave as it is_
Confetti img in normal mode: 1863x1577
Confetti img in football mode: 1170x1634
Medal img (normal mode): 223x309
Ball img (football mode): 239x235
Champagne img (normal mode): 499x629
Cup img (football mode): 593x617

** Options Page: **
_Client => Leave as it is_

ads_off img: 109x156
ads_on img: 108x156
sounds_off img: 111x85
sounds_on img: 106x97
stats_off img: 112x98
stats_on img: 103x98
toggle_off img: 258x143
toggle_on img: 249x138
help popup triangle img: use android default.

**Store Page:**
_Client => Leave as it is_

Football icon: 134x134
jokers icon (yellow): 134x135
jokers icon (grey): 134x135
no_ads icon: 134x135
premium_icon: 132x131

**Comment by ignacioch** [comment]
New link : https://www.dropbox.com/sh/yog7htmu6kxup4j/AADyoKZvEU8PmSn6SXpYT7e2a?dl=0

**Comment by panagiotissp** [comment]
@ignacioch  I'm trying to close this, but things do not make that much sense:

The usage of default toggle button is a big change, not simple at all. Is that 100% sure? 
The default android toggle button is this to support api 11+ as has been decided:
![toggle-api-11](https://user-images.githubusercontent.com/14311142/31738865-984ac574-b454-11e7-8a95-40e03eed3b25.PNG)
Moreover, there is no 'default popup window' to support the functionalities needed, in the different popups shown. (And a change there is also a big one). Popup WIndow is a functionality/widget in Android that you then setup dimensions, background color etc to it: https://developer.android.com/reference/android/widget/PopupWindow.html

I wrote that what is missing are the small triangle which is orange right now (a similar one in options page too) and the images instead of the 'book images' I used below:
![main-menu-popup](https://user-images.githubusercontent.com/14311142/31738998-1377d02a-b455-11e7-9efc-458ce7808cd9.PNG)

This is confusing so maybe a call would be better here

**Comment by ignacioch** [comment]
Default popup window I mean a toast message. 
https://developer.android.com/guide/topics/ui/notifiers/toasts.html

I dont think this has any complexity.

> The usage of default toggle button is a big change, not simple at all. Is that 100% sure?
> The default android toggle button is this to support api 11+ as has been decided:

Why is this a big change? For ios it was simpler since I didn't have to do anything with background etcs. Maybe for Android is different. I will ask the clients if you want and let you know.

**Comment by ignacioch** [comment]
Default popup window is only used when clicking to see info for the 'Jokers' or 'Football' mode. Nowhere else in the app.

**Comment by ignacioch** [comment]
For the small triangle, I have it on IOS so I can send it to you.

**Comment by panagiotissp** [comment]
Wait:
- This 'toast' popup is trivial to create. But this kind of popup is nowhere in the specs I know. All the popups in the app are in the same philosophy until now, to have them extend from the top of the screen to the button clicked an also has this little triangle.
There is e.g. a help button that triggers a popup in the Options page too. So, this will remain as it is, and for the other 2 (jokers and football mode) it needs to change?
Also, for this popup implementation, is there any specification of how long to show that? (otherwise I will put the default android value here too)


- For the toggle buttons, right now they are not toggle buttons, they are just Images let's say. Where everywhere in the code I replace the 'on' image with the 'off' image. And also there is custom variables to have the state of the 'button'. Toggle buttons have this state as objects, so I need refactoring there too (these are a couple of things that came to my mind right now of why it is not easy to change that)

No need to ask the customer about that. If customer agrees on how the toggle buttons will look (my screenshot above) and we are sure the change will not be ..changed again in the future, then I will proceed with using these default toggle buttons.

Waiting for your replies

**Comment by ignacioch** [comment]
Ok I need to check current status of the app, difficult to say without see the app in a device. But from what I undestand : 

1. Yes, the first spec said all popups were custom. Couple of months ago, this was changed - I think you should have an email/issue about it but it might be lost after all these months. The two '?' popups in the GameSettings needs to be replaced by default popup - in Android that's a toast message I think or a DialogBox even better.

2. Ok, then this might be faulty implementation from the beginning. if it is image, then user does not have the "slide/toggle" interaction/feeling. It will just click an image and the status changes. What I had in mind is that we use : https://developer.android.com/guide/topics/ui/controls/togglebutton.html - and if parameters like background colour is settable we set it as we do in ios.

will check the app on its current status.

**Comment by panagiotissp** [comment]
- I changed all toggle buttons to Android switches and specifically 'SwitchCompat' which is compatible for all target devices.
- I replaced the 2 help popups in Game Settings P2 with Toast messages. Set custom duration of 10 seconds for the Toast that opens when clicking the Locked jokers help button, because of the long text included.

@ignacioch  do these sound ok?
Should all the toggle buttons (not only the 2 in the Game Settings P2) changed to the default Android ones? This is what I did.
Should the help popup in the Options page be also replaced by the Toast? I did not change that since it is not mentioned in your comments above, but just a last check, since I find it weird, that we are aligned.

**Comment by ignacioch** [comment]
should be fine. I will check.

No on the options page I have theirs, and they didn't say to change it so we leave it as it is.

Issue can stay closed.

Sent from my phone, Sorry for typos.

On 24 Oct 2017, at 01:36, Panagiotis Spirou <notifications@github.com<mailto:notifications@github.com>> wrote:


  *   I changed all toggle buttons to Android switches and specifically 'SwitchCompat' which is compatible for all target devices.
  *   I replaced the 2 help popups in Game Settings P2 with Toast messages. Set custom duration of 10 seconds for the Toast that opens when clicking the Locked jokers help button, because of the long text included.

@ignacioch<https://github.com/ignacioch> do these sound ok?
Should all the toggle buttons (not only the 2 in the Game Settings P2) changed to the default Android ones? This is what I did.
Should the help popup in the Options page be also replaced by the Toast? I did not change that since it is not mentioned in your comments above, but just a last check, since I find it weird, that we are aligned.
